nimble/controller/src/ble_ll_conn.c

```diff
@@ -3907,7 +3907,8 @@ ble_ll_conn_rx_isr_end(uint8_t *rxbuf, struct ble_mbuf_hdr *rxhdr)
         }
 
         /* If we received a terminate IND, we must set some flags */
-        if (is_ctrl && (opcode == BLE_LL_CTRL_TERMINATE_IND)) {
+        if (is_ctrl && (opcode == BLE_LL_CTRL_TERMINATE_IND)
+                    && (rx_pyld_len == BLE_LL_CTRL_TERMINATE_IND_LEN)) {
```

Comment:
I think check for length should be in nested `if` since we should not go to `else` statements in this case

Reply:
As discussed off line, it make sense to do so